More esphome configuration

tagno25 opened the pull request:
I have added more esphome configuration setup and moved the serial info to a separate file.

Now there is a slider to set and show the desk height.
The switches have been changed to buttons, and more have been added.

Should the esphome files be stored as separate files or kept in the Serial.md file?

tagno25 committed:
- Added epshome sliders for desk height in Inches, CM, and raw
  - Custom componet required to interpret the received data
- Changed switches to buttons
- Added buttons for commands 11, 12, 14, 15, 17, and 18

tagno25 committed:
Moved serial info to separate file

philwmcdonald commented on Serial.md:
```diff
@@ -0,0 +1,312 @@
+<C0.0.        # reports current height
+<+300..       # nudge desk up by 300
+<L,5.         # recall position from slot 2
+<l,5.         # recall position from down-button slot 2
```
```suggestion
<L,2.         # recall position from slot 2
<l,2.         # recall position from down-button slot 2
```

philwmcdonald commented on Serial.md:
```diff
@@ -0,0 +1,312 @@
+      then:
+        - component.update: megadesk_height_inches
+        - component.update: megadesk_height_cm
+        - component.update: megadesk_height_raw
```
cm/inch heights - nice!

tagno25 replied:
Heights have NOT been verified to be accurate.  The min and max are based upon the heights stated by Ikea.

tagno25 replied:
Would it be possible for you measure what the true minimum and maximum heights are in either centimeters or inches?  My desk is on carpet, which would cause height inaccuracies.

gcormier replied:
On a large bekant I'm measuring

Lowest - 23" from floor to bottom of table surface
Highest - 47" from floor to bottom of table surface

philwmcdonald commented on Serial.md:
```diff
@@ -0,0 +1,312 @@
+      then:
+        - uart.write: "<L0,5."
+  - platform: template
+    name: "Set Minimum Desk Height"
```
Note. this toggles the min-height limit

philwmcdonald replied:
And you can always query the Min height with "<R0,11."

tagno25 replied:
Changed name from "Set" to "Toggle"

philwmcdonald commented on Serial.md:
```diff
@@ -0,0 +1,312 @@
+      then:
+        - uart.write: "<L0,11."
+  - platform: template
+    name: "Set Maximum Desk Height"
```
Note. this toggles the max-height limit

tagno25 replied:
Changed name from "Set" to "Toggle"

tagno25 committed:
- Fixed Serial.md examples
- esphome
  - Updated step size for inches and centimeters
  - Added sensors for minimum and maximum heights
  - Changed wording for minimum and maximum height buttons

tagno25 commented on Serial.md:
```diff
@@ -0,0 +1,331 @@
+        - component.update: megadesk_height_cm
+        - component.update: megadesk_height_raw
+  - name: "Megadesk Minimum Height"
+  - name: "Megadesk Maximum Height"
```
Should I add inch and centimeter sensors for the minimum and maximum height?  Or are just the raw numbers good enough?

gcormier commented:
I think now that the esphome is more then just a sample, we could break it out to its own `esphome.md` file. It would cut down on the noise for those who want serial control for other purposes.

tagno25 commented:
Should `esphome.md` be referenced in `README.md` or `Serial.md` or both?

gcormier commented:
I would say both?

tagno25 committed:
- Move epshome example to a seperate file
- changed esphome height range to verified heights

gcormier commented:
Awesome work! Would love a screenshot of it in HA with the slider if you can grab one.
